### ISSUE TITLE
Whitelist, Blacklist, And Between

### ISSUE THREAD

**Issue by almafeta** [almafeta/gallery#39]
Admins should be able to whitelist and blacklist galleries.  Incoming comments are permitted from whitelisted galleries, forbidden from blacklisted galleries.  In between the two, galleries need to behave moderately intelligently...

`alice.gallery` and `bob.gallery` are two known good actors; `mallory.gallery` and `vernon.gallery` are unknown actors.

`mallory.gallery` joins the federation and begins spamming `alice.gallery` and `bob.gallery`.  Alice suspects Mallory and sends out a notice to Bob.  However, Bob has Mallory whitelisted, because Bob knows Mallory out of band.  Alice keeps muting Mallory, but Bob keeps displaying Mallory.  `alice.gallery` notifies its admin.

Later, Mallory's behavior normalizes after the initial hubbub dies down.  `alice.gallery` notifies its admin that Mallory is still interacting and that their trusted gallery, `bob.gallery`, also trusts Mallory.  Alice whitelists `mallory.gallery` and all is well.

Later, `vernon.gallery` joins the federation and begins spamming `alice.gallery`, `bob.gallery`, and `mallory.gallery`.  Alice manually blacklists Vernon.  Bob manually whitelists.  Mallory is aware of this and can visit `vernon.gallery` to choose for themselves.

So in short...

1. Admins use can use recommendations from their whitelists to determine whether to preemptively whitelist or blacklist a new gallery - e.g., an admin might whitelist 5 galleries, preemptively blacklist galleries if 2 of their whitelisted galleries do so, and preemptively whitelist galleries if 3 of their whitelisted galleries do so.  Whitelist/blacklist entries generated this way are marked as preemptive.
2. Failing rule 1, moderators can preemptively whitelist or blacklist galleries.
3. Failing rules 1 and 2, behavior can be used to preemptively whitelist or blacklist galleries - e.g., if local users report a new gallery X times, it might be blacklisted; or if local users fave art from a gallery Y times, it might be whitelisted.

**Comment by almafeta** [comment]
tl;dr: assume other people are going to use the software to be a shit, make it easy to clean up after them.
